Add external link function and bad link text examples

cjrace opened the pull request:
EDITIED:

# Brief overview of changes

Added a custom wrapper for `htmltools::tags$a()` that makes external links easier to do and enforces accessibility.

## Why are these changes being made?

External links aren't generally done well by analysts, explaining how to do them is clunky and if not done correctly they can be insecure as well as inaccessible.

## Detailed description of changes

1. Added `external_link()` function.

In this I'm intentionally using a basic regex to check for URLs as wanted to avoid making requests every time the front end of a site using this is rendered, which would have happened if I'd used RCurl or httr's custom functions that check for a valid URL.

2. Added `bad_link_text` data frame with known examples of poor link text.

3. Reordered the reference list to be vaguely alphabetical (was also nicer to have cookies at the top instead of the mostly redundant tidy_code)

## Additional information for reviewers

Added a couple of extra commits since raising that remove excess whitespace, pre-emptively bump the package version, and also fix up the way links show in the documentation.

## Issue ticket number/s and link

No related issues, but does relate to some of the things flagged in the DAC audit.

cjrace commented:
Going to add a quick extra thing to enforce `.noWS = "after"` as we often end up needing that for neatness

cjrace commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Add external link function into dfeshiny](https://trello.com/c/poQLhjH9/1737-add-external-link-function-into-dfeshiny)

cjrace commented:
Went deeper into the roxygen2 documentation and with some experimentation have managed to get the links from the function and data documentation working nicer

rmbielby commented on R/external_link.R:
```diff
@@ -0,0 +1,102 @@
+  if (is_url(link_text)) {
+    stop(paste0(
+      link_text,
+      " has been recognise as a raw URL, please change the link_text value to",
```
Small typo: recognise -> recognised

rmbielby commented on R/external_link.R:
```diff
@@ -0,0 +1,102 @@
+  }
+
+  # Check against curated data set for link text we should banish into room 101
+  if (tolower(link_text) %in% dfeshiny::bad_link_text$bad_link_text) {
```
Also worth using `stringr::str_trim()` to remove leading and trailing white space from `link_text` just in case?

rmbielby commented on R/external_link.R:
```diff
@@ -0,0 +1,102 @@
+      )
+    )
+  }
+
```
Could also be worth adding some validation to make sure the link text doesn't end in a full stop - looks weird if there's a full stop and then the "opens in a new tab" in brackets

cjrace replied:
Good suggestion! Will add that

rmbielby reviewed (changes requested):
Few minor comments. I was looking at links on the HE LEO tab earlier so tested it out on a branch on that.

cjrace commented:
Addressed all of the comments from @rmbielby, added a couple more suggestions to the bad text list from Nusrath, and used Tom's suggestion of a blanket warning for particularly short strings in the link text argument.
